RoaringBitmap/src/main/java/org/roaringbitmap/longlong/Roaring64NavigableMap.java

```diff
@@ -7,1366 +7,1445 @@
-      BitmapDataProvider nextBitmap = highToBitmap.get(high);
-      return RoaringIntPacking.pack(high, nextBitmap.select(0));
-    } else {
-      // There is no bucket with this cardinality
-      int insertionPoint = -position - 1;
-
-      final long previousBucketCardinality;
-      if (insertionPoint == 0) {
-        previousBucketCardinality = 0L;
-      } else if (insertionPoint >= indexOk) {
-        return throwSelectInvalidIndex(j);
-      } else {
-        previousBucketCardinality = sortedCumulatedCardinality[insertionPoint - 1];
-      }
-
-      // We get a 'select' query for a single bitmap: should fit in an int
-      final int givenBitmapSelect = (int) (j - previousBucketCardinality);
-
-      int high = sortedHighs[insertionPoint];
-      BitmapDataProvider lowBitmap = highToBitmap.get(high);
-      int low = lowBitmap.select(givenBitmapSelect);
-
-      return RoaringIntPacking.pack(high, low);
-    }
-  }
-
-  // For benchmarks: compute without using cardinalities cache
-  // https://github.com/RoaringBitmap/CRoaring/blob/master/cpp/roaring64map.hh
-  private long selectNoCache(long j) {
-    long left = j;
-
-    for (Entry<Integer, BitmapDataProvider> entry : highToBitmap.entrySet()) {
-      long lowCardinality = entry.getValue().getCardinality();
-
-      if (left >= lowCardinality) {
-        left -= lowCardinality;
-      } else {
-        // It is legit for left to be negative
-        int leftAsUnsignedInt = (int) left;
-        return RoaringIntPacking.pack(entry.getKey(), entry.getValue().select(leftAsUnsignedInt));
-      }
-    }
-
-    return throwSelectInvalidIndex(j);
-  }
-
-  private long throwSelectInvalidIndex(long j) {
-    // see org.roaringbitmap.buffer.ImmutableRoaringBitmap.select(int)
-    throw new IllegalArgumentException(
-        "select " + j + " when the cardinality is " + this.getLongCardinality());
-  }
-
-  /**
-   * For better performance, consider the Use the {@link #forEach forEach} method.
-   *
-   * @return a custom iterator over set bits, the bits are traversed in ascending sorted order
-   */
-  public Iterator<Long> iterator() {
-    final LongIterator it = getLongIterator();
-
-    return new Iterator<Long>() {
-
-      @Override
-      public boolean hasNext() {
-        return it.hasNext();
-      }
-
-      @Override
-      public Long next() {
-        return it.next();
-      }
-
-      @Override
-      public void remove() {
-        // TODO?
-        throw new UnsupportedOperationException();
-      }
-    };
-  }
-
-  @Override
-  public void forEach(final LongConsumer lc) {
-    for (final Entry<Integer, BitmapDataProvider> highEntry : highToBitmap.entrySet()) {
-      highEntry.getValue().forEach(new IntConsumer() {
-
-        @Override
-        public void accept(int low) {
-          lc.accept(RoaringIntPacking.pack(highEntry.getKey(), low));
-        }
-      });
+    // Not final to enable initialization in Externalizable.readObject
```

Comment:
@baibaichen Please fix the indentation in your PR, it makes it very difficult to read as it stands